net/dns/resolver/forwarder.go

```diff
@@ -335,40 +336,43 @@ func (f *forwarder) packetListener(ip netaddr.IP) (packetListener, error) {
 func (f *forwarder) getKnownDoHClient(ip netaddr.IP) (urlBase string, c *http.Client, ok bool) {
 	urlBase, ok = publicdns.KnownDoH()[ip]
 	if !ok {
-		return
+		return "", nil, false
+	}
+	c, ok = f.getKnownDoHClientForProvider(urlBase)
+	if !ok {
+		return "", nil, false
 	}
+	return urlBase, c, true
+}
 
+func (f *forwarder) getKnownDoHClientForProvider(urlBase string) (c *http.Client, ok bool) {
```

Comment:
this split is so the `ip` variable can't be accidentally used in the code below and that it's clear that the `*http.Client` returned is purely for the `urlBase`.

Reply:
add a comment so that it doesn't regress

Reply:
done.